FlowCrypt/Controllers/Setup/SetupCreatePassphraseViewController.swift

```diff
@@ -188,6 +195,48 @@ extension SetupGenerateKeyViewController {
         }
     }
 
+    private func setupAccountWithExistingKeys(with passPhrase: String) {
+        Promise { [weak self] in
+            guard let self = self else { return }
+            self.showSpinner()
+
+            try awaitPromise(self.validateAndConfirmNewPassPhraseOrReject(passPhrase: passPhrase))
+
+            try self.keys.forEach { key in
+                try key.keyDetails.forEach { keyDetail in
+                    guard let privateKey = keyDetail.private else {
+                        throw CreatePassphraseWithExistingKeyError.noPrivateKey
+                    }
+                    let encryptedPrv = try self.core.encryptKey(
+                        armoredPrv: privateKey,
+                        passphrase: passPhrase
+                    )
+                    let parsedKey = try self.core.parseKeys(armoredOrBinary: encryptedPrv.encryptedKey.data())
+                    parsedKey.keyDetails.forEach { keyDetails in
+                        let passPhrase = PassPhrase(value: passPhrase, fingerprints: keyDetails.fingerprints)
+                        self.passPhraseService.savePassPhrase(with: passPhrase, inStorage: self.shouldSaveLocally)
+                    }
+                    try awaitPromise(self.backupService.backupToInbox(keys: parsedKey.keyDetails, for: self.user))
+                    self.keyStorage.addKeys(keyDetails: parsedKey.keyDetails, source: .generated, for: self.user.email)
```

Comment:
The `source` here is wrong, the key was not generated in the app.

Reply:
Should I add a new case for `source`?

Reply:
Yes, let's do that.